app/src/main/kotlin/com/exactpro/th2/lwdataprovider/http/DownloadEventsHandler.kt

```diff
@@ -0,0 +1,154 @@
+import com.exactpro.th2.lwdataprovider.handlers.SearchEventsHandler
+import com.exactpro.th2.lwdataprovider.http.util.JSON_STREAM_CONTENT_TYPE
+import com.exactpro.th2.lwdataprovider.http.util.writeJsonStream
+import com.exactpro.th2.lwdataprovider.workers.KeepAliveHandler
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.javalin.Javalin
+import io.javalin.http.Context
+import io.javalin.http.queryParamAsClass
+import io.javalin.openapi.HttpMethod
+import io.javalin.openapi.OpenApi
+import io.javalin.openapi.OpenApiContent
+import io.javalin.openapi.OpenApiParam
+import io.javalin.openapi.OpenApiResponse
+import java.time.Instant
+import java.util.concurrent.ArrayBlockingQueue
+import java.util.concurrent.Executor
+import java.util.function.Supplier
+
+class DownloadEventsHandler(
+    private val configuration: Configuration,
+    private val convExecutor: Executor,
+    private val sseResponseBuilder: SseResponseBuilder,
+    private val keepAliveHandler: KeepAliveHandler,
+    private val searchEventsHandler: SearchEventsHandler,
+    private val dataMeasurement: DataMeasurement,
+) : JavalinHandler {
+    override fun setup(app: Javalin, context: JavalinContext) {
+        app.get(ROUTE_DOWNLOAD_EVENTS, this::handleMessage)
+    }
+
+    @OpenApi(
+        path = ROUTE_DOWNLOAD_EVENTS,
+        description = "returns list of messages for specified groups. Each group will be requested one after another " +
+                "(there is no order guaranties between groups). Messages for a group are not sorted by default. ",
+        queryParams = [
+            OpenApiParam(START_TIMESTAMP_PARAM, type = Long::class, required = true, example = HttpServer.TIME_EXAMPLE,
+                description = "start timestamp for search. Epoch time in milliseconds"),
+            OpenApiParam(END_TIMESTAMP_PARAM, type = Long::class, required = true, example = HttpServer.TIME_EXAMPLE,
+                description = "end timestamp for search. Epoch time in milliseconds"),
+            OpenApiParam(PARENT_EVENT_PARAM, type = String::class,
+                description = "parent event id for search", example = "testEventId123"),
+            OpenApiParam(KEEP_OPEN_PARAMETER, type = Boolean::class,
+                description = "enables pulling for updates until have not found a message outside the requested interval"),
+            OpenApiParam(BOOK_ID_PARAM, required = true, example = "bookId123",
+                description = "book ID for requested groups"),
+            OpenApiParam(SCOPE_PARAM, type = String::class, required = true,
+                description = "scope for events", example = "scope123"),
+            OpenApiParam(LIMIT, type = Int::class,
+                description = "limit for events in the response. No limit if not specified"),
+            OpenApiParam(SEARCH_DIRECTION, type = SearchDirection::class, example = "next",
+                description = "defines the order of the events"),
```

Comment:
issue: The description does not match the reality. Please correct them before this can be merged

Reply:
Do you mean `keepOpen` parameter ?

Reply:
I'd appreciate it if you could read all the descriptions and you will see what is wrong. You copy-pasted it from messages - there are messages mentioned in the descriptions

Reply:
Corrected